src/wp-includes/blocks/index.php

```diff
@@ -40,26 +39,62 @@ function register_core_block_style_handles() {
 		'editorStyle' => 'editor',
 	);
 
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require $includes_path . 'blocks/blocks-json.php';
+	}
+
+	$files          = false;
+	$transient_name = 'wp_core_block_css_files';
 	/*
 	 * Ignore transient cache when the development mode is set to 'core'. Why? To avoid interfering with
 	 * the core developer's workflow.
 	 */
 	if ( ! wp_is_development_mode( 'core' ) ) {
-		$transient_name = 'wp_core_block_css_files';
-		$files          = get_transient( $transient_name );
-		if ( ! $files ) {
-			$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
-			set_transient( $transient_name, $files );
+		$cached_files = get_transient( $transient_name );
+
+		/*
+		 * Check the validity of cached values.
+		 *  - Should match the current WordPress version
+		 *  - Should use relative paths for the files
+		 */
```

Comment:
See my other comment, the relative file check isn't necessary any more since we already check the version which will prevent the older caches to ever be used.
```suggestion
		// Check the validity of cached values by checking against the current WordPress version.
```